app/main.py

```diff
@@ -1,8 +1,26 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.spouse = None
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+        spouse_name = person_data.get("wife") or person_data.get("husband")
+        if spouse_name:
+            spouse = Person.people.get(spouse_name)
+            if spouse:
```

Comment:
Shorten the logic of these if conditions by combining them into one and avoiding additional nesting:
```suggestion
        if spouse_name and (spouse := Person.people.get(spouse_name)):
```